doc/TEMPLATES.md

```diff
@@ -67,6 +67,20 @@ The following template languages are supported:
 * html
 * svg
 
+### Nested Data and Re-rendering
+
+Web components will only trigger a re-render by default **when their properties are replaced**. This means that when using non-primitive data structures as properties, mutating the contents of these structures will not result in the component re-rendering. You must either replace the data structure or manually call the `.render` method.
```

Comment:
My high-level feedback is that the `x-element` documentation ought to advocate for _consistently_ leveraging change-by-reference detection patterns.

I.e., rather than say “if you use a compound object, you need to do extra imperative calls when you mutate the data” — we should say “mint new object references versus mutating data in place”.

Object / array spread operators make this pretty straightforward in modern JS. The bet is that the cost (both performance and complexity) is cheaper to simply update object references when binding properties (versus encoding and performing many imperative calls to mutate DOM).

Reply:
@theengineear I've updated the docs to avoid mentioning the imperative re-rending method, and have made it clear that change-by-reference is recommended.